### ISSUE TITLE
Use environment variables under inlineScrapeConfig

### ISSUE THREAD

**Issue by Angel0r** [VictoriaMetrics/helm-charts#1047]
Hello,

I am using `victoria-metrics-k8s-stack` Helm chart in our Kubernetes production cluster. 
I am trying to configure vmagent so it can scrape node-exporter metrics from our ClickHouse hosts which are external. The node-exporter runs with tls and basic auth.
My problem is that I cannot pass the basicauth creds(username and password) into inlineScrapeConfigs. Here is my configuration:
```yaml
    vmagent:
      enabled: true
      annotations: {}
      # https://docs.victoriametrics.com/operator/api.html#vmagentremotewritespec
      # defined spec will be added to the remoteWrite configuration of VMAgent
      additionalRemoteWrites:
        []
        #- url: http://some-remote-write/api/v1/write
      # spec for VMAgent crd
      # https://docs.victoriametrics.com/operator/api.html#vmagentspec
      spec:
        inlineScrapeConfig: |

          - job_name: clickhouse-nodes-monitor
            scheme: https
            tls_config:
              cert_file: /etc/vmagent-tls/certs/cert.crt
              key_file: /etc/vmagent-tls/certs/key.key
            basic_auth:
              username: ${username}
              password: ${password}
            static_configs:
              - labels:
                  job: clickhouse-nodes
                targets:
                  - "clickhouse01:9100"
                  - "clickhouse02:9100"
                  - "clickhouse03:9100"
                  - "clickhouse04:9100"

        containers:
          - name: vmagent
            volumeMounts:
              - name: vmagent-tls-certs
                mountPath: /etc/vmagent-tls/certs
            envFrom:
              - secretRef:
                  name: clickhouse-certs

        volumes:
          - name: vmagent-tls-certs
            secret:
              secretName: clickhouse-certs
```

I have checked the creds in the secret and they are correct.
I have also tried `extraEnvs` but still I get `Unauthorized`.
What am I missing?

/Angelos

**Comment by Haleygo** [comment]
Hello!
How are `${username}` and `${password}` being passed here, do you defined them in pod or in chart value file?

**Comment by Angel0r** [comment]
Hello Haleygo,

Thank for you answer!
I have a secret and then I use `envFrom` under containers:
```yaml
        containers:
          - name: vmagent
            volumeMounts:
              - name: vmagent-tls-certs
                mountPath: /etc/vmagent-tls/certs
            envFrom:
              - secretRef:
                  name: clickhouse-certs
```

**Comment by Angel0r** [comment]
I added them as `extrEnvs` under `vmagent.spec` and I pass them as you said with the `%` and it works fine!
```yaml
      spec:
        extraEnvs:
          - name: USERNAME
            valueFrom:
              secretKeyRef:
                name: clickhouse-certs
                key: username
          - name: PASSWORD
            valueFrom:
              secretKeyRef:
                name: clickhouse-certs
                key: password

        inlineScrapeConfig: |

          - job_name: clickhouse-nodes-monitor
            scheme: https
            tls_config:
              cert_file: /etc/vmagent-tls/certs/cert.crt
              key_file: /etc/vmagent-tls/certs/key.key
            basic_auth:
              username: %{USERNAME}
              password: %{PASSWORD}
            static_configs:
              - labels:
                  job: clickhouse-nodes
                targets:
                  - "clickhouse01:9100"
                  - "clickhouse02:9100"
                  - "clickhouse03:9100"
                  - "clickhouse04:9100"
```
Sorry I missed the `%` in the documentation!
Thanks!

/Angelos

**Comment by Angel0r** [comment]
Hey @Haleygo,
It doesn't work. I mistaken since I tested wrong configuration(with password_file)...
It seems like we cannot add variable inside `inlineScrapeConfigs`.
Is that true?
When I added the username and password as is it worked.
Can we re-open the issue?

/Angelos

**Comment by Haleygo** [comment]
>When I added the username and password as is it worked.

You mean `%{ENV_VAR}` works only for `username` and `password` or doesn't work at all? Could you share your config?

**Comment by Angel0r** [comment]
@Haleygo sorry the delay. I am still testing.
No I meant if I don't use variables and use the `username` and `password` directly then it works. 
My configuration is the same as I showed above in my latest comment.

When I suspend flux and make changes directly in helmrelease to test the variables I get weird behaviour. The VMAgent doesn't redeploy for example when I add a new `inlineScrapeConfig` with tls and basicauth. If I remove the tls/basicauth then VMAgent is redeployed and get errors like `cannot scrape...` but in Grafana the logs start to be showing :exploding_head: 

I saw in the documentation there is a note saying "Do not use passwords and tokens..."
https://docs.victoriametrics.com/operator/resources/vmagent/#inline-additional-scrape-configuration-in-vmagent-crd
This note is just for security reasons or functionality reasons?
